### PR TITLE
Macros IS_GUEST_KEY

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

This PR simply macros an old Proc that checks if a key is a guest key or not by simply making it a string comparison maybe still not the best solution but better than the original approach definitly.
Now this also makes the macro with 2 inputs for a very simple reason the macro is used in wildly different enviroments most do have a client datum but not all of them so i opted to simply get the var here directly and individually to the usecase.
## Why It's Good For The Game
one global proc less

</details>

## Changelog
:cl:
refactor: Refactored guest player detection to be much simpler.
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
This *seems* fine but it could still be sus. I kind of wonder how this affects CID spoofers.

**Comment by MNarath1** [comment]
> This _seems_ fine but it could still be sus. I kind of wonder how this affects CID spoofers.

considering that key's probably are send to the server (not perfectly sure on that) this may be a problem i mean it would not be much of an issue if the key is generated server side but a big one if its client side cause then someone just needs to change either their cid afterwards for this to not work anymore

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -11,6 +11,9 @@
 
 #define isweakref(D) (istype(D, /datum/weakref))
 
+// simple check whether or not a player is a guest using their key and computer_id to build the key and then make a string comparison with the actual key
+#define IS_GUEST_KEY(key, computer_id)	("[key]" == "Guest-[computer_id]")
```
Comment (ike709) [inline code/__DEFINES/is_helpers.dm]: After pondering it, to stay on the safe side I think you should check if it starts with `Guest-` and then if the remainder of it is a number.

Comment (ike709) [reply code/__DEFINES/is_helpers.dm]: Or just only check that it starts with `Guest-`

Comment (ike709) [reply code/__DEFINES/is_helpers.dm]: Yeah just check if it starts with `Guest-`

File: code/__DEFINES/is_helpers.dm [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 #define isweakref(D) (istype(D, /datum/weakref))
 
+// simple check whether or not a player is a guest using their key
+#define IS_GUEST_KEY(key)	(findtext(key, "Guest-", 1, 7))
```
Comment (ike709) [inline code/__DEFINES/is_helpers.dm]:
```suggestion
#define IS_GUEST_KEY(key)	(findtextEx(key, "Guest-", 1, 7))
```

**Review by ike709 — APPROVED** [review]

**Pull request merged**